delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/Storages.scala

```diff
@@ -0,0 +1,110 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage
+
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.StorageCommand.{CreateStorage, DeprecateStorage, TagStorage, UpdateStorage}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.StorageEvent.{StorageCreated, StorageDeprecated, StorageTagAdded, StorageUpdated}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.StorageRejection._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.StorageState.{Current, Initial}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.{StorageCommand, StorageEvent, StorageRejection, StorageState, StorageValue}
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils
+import monix.bio.{IO, UIO}
+
+class Storages {}
+
+object Storages {
+
+  /**
+    * The storages module type.
+    */
+  final val moduleType: String = "storage"
+
+  private[storage] type StorageAccess = StorageValue => IO[StorageNotAccessible, Unit]
+
+  private[storage] def next(
+      state: StorageState,
+      event: StorageEvent
+  ): StorageState = {
+    // format: off
+    def created(e: StorageCreated): StorageState = state match {
+      case Initial     => Current(e.id, e.project, e.value, e.source, Map.empty, e.rev, deprecated = false,  e.instant, e.subject, e.instant, e.subject)
+      case s: Current  => s
+    }
+
+    def updated(e: StorageUpdated): StorageState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, value = e.value, source = e.source, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    def tagAdded(e: StorageTagAdded): StorageState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, tags = s.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+    }
+    // format: on
+
+    def deprecated(e: StorageDeprecated): StorageState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    event match {
+      case e: StorageCreated    => created(e)
+      case e: StorageUpdated    => updated(e)
+      case e: StorageTagAdded   => tagAdded(e)
+      case e: StorageDeprecated => deprecated(e)
+    }
+  }
+
+  private[storage] def evaluate(
```

Comment:
When we already have a default storage for a project, shouldn't we fail ?

Reply:
This logic, as for when we have already a resolver with the same priority in a project, cannot be handled in the state machine

Reply:
I haven't done it with the priority thing with resolvers because we said it could be done later.
For realms for exemple, we check for duplicates in the STM.

Reply:
It does not apply anymore. As we decided, this will be done in the `Storages` logic synchronously.